sites/all/modules/reol_app_feeds/src/Helper/ParagraphHelper.php

```diff
@@ -426,22 +426,25 @@ public function getThemeData($node) {
       $image = !empty($node->field_breol_cover_image) ? $this->nodeHelper->getImage($node->field_breol_cover_image) : static::VALUE_NONE;
 
       // Get identifiers from carousel queries.
-      $carousels = $this->nodeHelper->getFieldValue($node, 'field_carousels', NULL, TRUE);
       $identifiers = [];
-      foreach ($carousels as $carousel) {
-        module_load_include('inc', 'opensearch', 'opensearch.client');
-        $result = opensearch_do_search($carousel['search']);
-        foreach ($result->collections as $collection) {
-          /** @var \TingCollection $collection */
-          /** @var \TingEntity $object */
-          $object = $collection->getPrimary_object();
-          $identifier = $object->getId();
-          if (!in_array($identifier, $identifiers)) {
-            $identifiers[] = $identifier;
+      $carousels = $this->nodeHelper->getFieldValue($node, 'field_carousels', NULL, TRUE);
+      if (is_array($carousels)) {
+        foreach ($carousels as $carousel) {
+          module_load_include('inc', 'opensearch', 'opensearch.client');
+          // Load at most 100 results.
+          $result = opensearch_do_search($carousel['search'], 1, 100);
+          foreach ($result->collections as $collection) {
+            /** @var \TingCollection $collection */
+            /** @var \TingEntity $object */
+            $object = $collection->getPrimary_object();
+            $identifier = $object->getId();
+            if (!in_array($identifier, $identifiers)) {
+              $identifiers[] = $identifier;
+            }
           }
+          // Only get identifiers from the first query.
+          break;
```

Comment:
There is no checks if the first carousels has any search results (and go to next if it has not), so why do it with `foreach` loop and not just do an `$carousel = reset($carousels);`